### PR TITLE
COL-737, link names on asset detail to profile if target tool is available

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-737

`crossLtiSupport` (i.e., linking between LTI tools) is now a shared utility. Future cross-LTI actions might include linking to assets from profile page.

### CODE REVIEW

**Comment by paulkerschen** [comment]
@johncrossman This makes sense, but two questions on naming:
1) Will this pattern be used for routing within the dashboard tool (e.g. navigating between profiles) as well as routing between LTI tools? If so, `crossLtiSupport` might be a misleading name.
2) `provider` seems like an opaque name for the current course, and I didn't quite follow why `crossLtiSupport` needs to return it if it's the same argument with `crossLtiSupport` is initialized. Is there any reason not to just use the `$scope.course` or `$scope.me.course` we have available?

**Comment by johncrossman** [comment]
@paulkerschen 
1. No, it was not my intention to use this "support" to navigate _within_ the new tool. The hope is that other cross-LTI scenarios (e.g., linking to asset from profile) would use same `crossLtiSupport()`
2. I agree on the opaque naming – amendment coming. And yes to `$scope.me.course`! Who knew?!

File: public/app/shared/utilService.js [hunk 1]
```diff
@@ -89,29 +89,6 @@
...
-     * @param  {Object}       course          Course with one or more SuiteC tools
-     * @param  {Object}       user            Member of course
-     * @return {String}                       URL with necessary query args
-     */
-    var getUserDashboardUrl = function(course, user) {
-      var dashboardUrl = getGenericDashboardUrl(course);
-      return dashboardUrl && user ? dashboardUrl + '?_id=' + user.id : null;
-    };
-
-    /**
```
Comment (johncrossman) [inline public/app/shared/utilService.js]: @paulkerschen The above logic was moved to a directive. Appropriate? Better? I'll let you be the judge.

File: public/app/shared/toolHrefDirective.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+              return course.whiteboards_url;
+            }
+            default: {
+              return '';
+            }
+          }
+        };
+        var queryArgs = scope.id ? '?_id=' + scope.id : '';
+        elem.attr('href', getToolUrl(scope.course, attrs.tool) + queryArgs);
+        elem.attr('target', '_parent');
```
Comment (johncrossman) [inline public/app/shared/toolHrefDirective.js]: Our new LTI tool will link to most/all other SuiteC tools.

**Comment by paulkerschen** [comment]
I think the directive is sweet, @johncrossman. Thanks!

**Pull request merged**